src/components/atoms/ButtonBuy.tsx

```diff
@@ -115,6 +109,18 @@ function getComputeAssetHelpText(
   return computeHelpText
 }
 
+function getAlgorithmConsumableStatusHelpText(
+  algorithmConsumableStatus: number
+) {
+  return algorithmConsumableStatus === 1
+    ? 'The selected algorithm has been temporarily disabled by the publisher, please try again later.'
+    : algorithmConsumableStatus === 2
+    ? 'Access denied, your wallet address is not found on the selected algorithm allow list.'
+    : algorithmConsumableStatus === 3
+    ? 'Access denied, your wallet address is found on the selected algorithm deny list.'
+    : ''
```

Comment:
nitpick(blocking): let's have a simple default message here, in case any new numbers are added, so the user does not get an empty alert with no message.
Something like "The selected algortihm is currently not available. You may try again later."